packages/transformers/typescript-types/src/TSTypesTransformer.js

```diff
@@ -155,6 +155,8 @@ export default (new Transformer({
     return [
       {
         type: 'ts',
+        // Stay on the types pipeline, even if the type changes
+        pipeline: asset.pipeline,
```

Comment:
Due to a bug in Transformation, this never surfaced, but changing an asset type from `tsx` to `ts` would also reset the pipeline, and this Typescript asset would then continue on the regular js pipeline.